lib/pure/sugar.nim

```diff
@@ -231,9 +231,15 @@ macro capture*(locals: varargs[typed], body: untyped): untyped {.since: (1, 1).}
   let locals = if locals.len == 1 and locals[0].kind == nnkBracket: locals[0]
                else: locals
   for arg in locals:
-    if arg.strVal == "result":
+    var argStr = ""
+    case arg.kind
+    of nnkIdent, nnkSym: argStr = arg.strVal
+    of nnkHiddenDeref: argStr = arg[0].strVal
```

Comment:
You don't know that the inner node is nnkIdent or nnkSym. What about `var seq[int]`?

Reply:
Fair.

Reply:
Well, in the particular case of `var seq[int]`, it's still the same AST (apparently), so `arg[0]` seems valid. But I'm pushing a small fix.